sdk/eventhub/azure-eventhub/azure/eventhub/aio/_connection_manager_async.py

```diff
@@ -62,20 +69,20 @@ def __init__(self, **kwargs):
 
 
 class _SeparateConnectionManager(object):
-    def __init__(self, **kwargs):
+    def __init__(self, **kwargs) -> None:
         pass
 
-    async def get_connection(self, host, auth):
+    async def get_connection(self, host: str, auth: 'JWTTokenAsync') -> None:
         pass  # return None
 
-    async def close_connection(self):
+    async def close_connection(self) -> None:
         pass
 
-    async def reset_connection_if_broken(self):
+    async def reset_connection_if_broken(self) -> None:
         pass
 
 
-def get_connection_manager(**kwargs):
+def get_connection_manager(**kwargs) -> Union[_SeparateConnectionManager, _SharedConnectionManager]:
```

Comment:
That probably means you need a BaseConnectionManager to declare the API common in both

Reply:
I added ConnectionManager as the base class and removed Union here.
A side note, should we avoid using duck-typing to have more mypy friendly code? Python is a dynamic typing language. By adding type hints and mypy rules, it's more like a static typing language. In this example, the newly added ConnectionManager acts like an interface in Java.

Reply:
If your base class doesn't have any code at all that could be shared, then the best approach would be a protocol:
https://github.com/Azure/azure-sdk-for-python/blob/ebd73c3fc22dcf17be2a903f32bdd95d9090f283/sdk/core/azure-core/azure/core/credentials.py#L15-L24

Reply:
Changed to use Protocol